modules/setup/datamodule.py

```diff
@@ -68,17 +68,21 @@ async def load_btc_drawdown(self, df: dict):
         bitcoin_drawdown = get_max_drawdown_ratio(bitcoin_values)
         return bitcoin_drawdown
 
-    async def load_historical_data(self) -> dict:
-        dataframes = await asyncio.gather(*[self.get_pair_data(pair) for pair in self.config.pairs])
+    async def load_historical_data(self, pairs, check_period=True) -> dict:
+        dataframes = await asyncio.gather(*[self.get_pair_data(pair, self.config.timeframe) if not isinstance(pair, tuple)
+                                            else self.get_pair_data(pair[0], pair[1]) for pair in pairs])   # if tuple then additional pair and timeframe comes specified with it
 
         history_data = {key: value for [key, value] in dataframes}
 
         self.warn_if_missing_ticks(history_data)
-        if not is_same_backtesting_period(history_data):
+        if not is_same_backtesting_period(history_data) and check_period:
```

Comment:
```suggestion
        if check_backtesting_period and not is_same_backtesting_period(history_data):
```
The first one will get checked first, so if it's false the code doesn't have to do the is_same_backtesting_period function! Also I'd rename check_period to check_backtesting_period, just to make it a tad clearer!

Reply:
Good one, thanks!